docs/architecture/adr/ADR044-security-improvements.md

```diff
@@ -0,0 +1,113 @@
+# ADR044: Security improvements
+
+## Status
+
+Pending
+
+## Context
+
+Currently there are no restrictions on what options a service team
+puts into pod specifications.  This means that a pod in a tenant
+namespace could escalate its privileges to access resources that it
+should not have access to.
+
+Examples include:
+
+* host namespaces
+* privileged containers
+* host networking
+* arbitrary host volume mounts
+
+For a full list see [pod security policies][] in the kubernetes
+documentation.
+
+Currently, our defence against this is that we enforce deploying
+infrastructure as code, and we require 2-eyes review on every code
+change.  However, kubernetes offers pod security policies as an extra
+layer of defence.
+
+[pod security policies]: https://kubernetes.io/docs/concepts/policy/pod-security-policy/
+
+## Decision
+
+We will implement pod security policies according to the following:
+
+### Cluster defaults
+
+```
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: cluster-default
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
+spec:
+  privileged: false
+  allowPrivilegeEscalation: true
+  allowedCapabilities:
+  - '*'
+  # Allow core volume types.
+  volumes:
+    - 'configMap'
+    - 'emptyDir'
+    - 'projected'
+    - 'secret'
+    - 'downwardAPI'
+    # Assume that persistentVolumes set up by the cluster admin are safe to use.
+    - 'persistentVolumeClaim'
+  hostNetwork: false
+  hostIPC: false
+  hostPID: false
+  runAsUser:
+    rule: 'RunAsAny'
+  seLinux:
+    rule: 'RunAsAny'
+  supplementalGroups:
+    rule: 'RunAsAny'
+  fsGroup:
+    rule: 'RunAsAny'
+  readOnlyRootFilesystem: false
+```
+
+### System namespaces
+
+For system namespaces, including `gsp-system`, `kube-system` and `istio-system`,
+should use a less restrictive policy.
+
+```
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: system
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
+spec:
+  privileged: true
+  allowPrivilegeEscalation: true
+  allowedCapabilities:
+  - '*'
+  volumes:
+  - '*'
+  hostNetwork: true
+  hostPorts:
+  - min: 0
+    max: 65535
+  hostIPC: true
```

Comment:
Don't think we have anything needing this do we?

Reply:
Probably not but I didn't want to risk bringing in subtle bugs to the system namespaces at this point, given the focus is securing the tenant namespaces.